src/components/deploymentConfig/ChartSelectorDropdown.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from 'react'
+import { PopupMenu, RadioGroup, sortObjectArrayAlphabetically, versionComparator, VisibleModal } from '../common'
+import { ReactComponent as Dropdown } from '../../assets/icons/ic-chevron-down.svg'
+import { ChartSelectorModalType } from './types'
+import { SortingOrder } from '../app/types'
+import { CHART_DOCUMENTATION_LINK, CHART_TYPE_TAB, CHART_TYPE_TAB_KEYS, RECOMMENDED_CHART_NAME } from './constants'
+
+export default function ChartSelectorDropdown({
+    charts,
+    chartsMetadata,
+    selectedChartRefId,
+    selectedChart,
+    selectChart,
+    isUnSet,
+}: ChartSelectorModalType) {
+    const [popupOpen, togglePopup] = useState(false)
+    const [selectedChartTypeTab, setSelectedChartTypeTab] = useState(
+        selectedChart?.['userUploaded'] ? CHART_TYPE_TAB_KEYS.CUSTOM_CHARTS : CHART_TYPE_TAB_KEYS.DEVTRON_CHART,
+    )
+    const uniqueChartsByDevtron = new Map<string, boolean>(),
+        uniqueCustomCharts = new Map<string, boolean>()
+    let devtronCharts = [],
+        customCharts = []
+
+    for (let chart of charts) {
+        const chartName = chart.name
+        if (chart['userUploaded']) {
+            if (!uniqueCustomCharts.get(chartName)) {
+                uniqueCustomCharts.set(chartName, true)
+                customCharts.push(chart)
+            }
+        } else if (!uniqueChartsByDevtron.get(chartName)) {
+            uniqueChartsByDevtron.set(chartName, true)
+            devtronCharts.push(chart)
+        }
+    }
+    devtronCharts = sortObjectArrayAlphabetically(devtronCharts, 'name')
+    customCharts = sortObjectArrayAlphabetically(customCharts, 'name')
+
+    const onSelectChartType = (selectedChartName: string): void => {
+        const filteredCharts = charts.filter((chart) => chart.name == selectedChartName)
+        const selectedChart = filteredCharts.find((chart) => chart.id == selectedChartRefId)
+        if (selectedChart) {
+            selectChart(selectedChart)
+        } else {
+            const sortedFilteredCharts = filteredCharts.sort((a, b) =>
+                versionComparator(a, b, 'version', SortingOrder.DESC),
+            )
+            selectChart(sortedFilteredCharts[sortedFilteredCharts.length ? sortedFilteredCharts.length - 1 : 0])
+        }
+    }
+
+    const changeSelectedTab = (event): void => {
+        setSelectedChartTypeTab(event.target.value)
+    }
+
+    const stopPropagation = (event): void => {
```

Comment:
Suggestion: Can we keep this function in common? As this is generic and can be used in the future aspect

Reply:
Sure